code/utils/torch_color_describer.py

```diff
@@ -242,6 +244,7 @@ def forward(self, word_seqs, seq_lengths=None, hidden=None, target_colors=None):
                 enforce_sorted=False)
             # RNN forward:
             output, hidden = self.rnn(embs, hidden)
+            output = self.dropout(output)
```

Comment:
We pass the `output` of the rnn to the dropout layer. This is why the encoder does not need a dropout layer as we only pass the hidden states to the decoder.